kumascript/macros/PWASidebar.ejs

```diff
@@ -0,0 +1,117 @@
+<%
+
+const sections = [
+  {
+    name: "Guides",
+    contents: [
+      "Guides/What_is_a_PWA",
+      "Guides/Benefits_of_PWAs",
+      {
+        name: "Guides/Main_features_of_PWAs",
+        contents: [
+          "Guides/Main_features_of_PWAs/Installability",
+          "Guides/Main_features_of_PWAs/Offline_operation",
+        ],
+      },
+    ],
+  },
+  {
+    name: "Tutorials",
+    contents: [
+      {
+        name: "Tutorials/Your_first_PWA",
+        contents: [
+          "Tutorials/Your_first_PWA/First_step",
+          "Tutorials/Your_first_PWA/Next_step",
+        ],
+      },
+      {
+        name: "Tutorials/Your_second_PWA",
+        contents: [
+          "Tutorials/Your_second_PWA/First_step",
+          "Tutorials/Your_second_PWA/Next_step",
+        ],
+      },
+    ],
+  },
+  {
+    name: "How-to",
+    contents: [
+      "How-to/Handle_app_updates",
+      "How-to/Implement_a_custom_install_flow",
+    ],
+  },
+  {
+    name: "Reference",
+    contents: [],
+  },
+];
+
+const sidebarURL = `/docs/Web/Progressive_web_apps/`;
+const baseURL = `/${env.locale}${sidebarURL}`;
+
+async function getTitle(pageSlug) {
+  let page = await wiki.getPage(`${baseURL}${pageSlug}`);
+  if (!page.title) {
+    page = await wiki.getPage(`/en-US${sidebarURL}${pageSlug}`);
+  }
+  return page.title;
+}
+
+async function renderSubsection(subsection) {
+  let output = `<li><details><summary>${await getTitle(
+    subsection.name
+  )}</summary><ol>`;
+
+  for (const item of subsection.contents) {
+    output += await renderItem(item);
+  }
+  output += "</ol></details></li>";
+
+  return output;
+}
+
+async function renderItem(item) {
+  let output = "";
+  if (typeof item === "string") {
+    output += `<li><a href="${baseURL}${item}">${await getTitle(
+      item
+    )}</a></li>`;
```

Comment:
Probably makes sense to extract `renderLink()`, which determines both the `href` and the text.

If the page does not exist in the current locale, I think we'll want to add `(en-US)` behind, like `web.smartlink()` already does (I don't mind not using `web.smartlink()` for now, but we should extract all these helper methods somewhere before replicating them to another sidebar anyways).